File: RepoTransBench/target_projects/Python/C++/underyx_flask-redis/public_tests/unit/test_public_init_py_test.cpp
#include <gtest/gtest.h>
#include <string>
#include <vector>
#include <regex>

namespace flask_redis_metadata_public {
    const std::string __version__ = "2.0.1";
    const std::string __title__ = "Flask-Redis";
    const std::string __description__ = "Yet another redis extension for Flask web apps.";
    const std::string __url__ = "https://flask-redis.public";
    const std::string __uri__ = __url__;
    const std::string __author__ = "Somebody";
    const std::string __email__ = "somebody@flaskredis.com";
    const std::string __license__ = "BSD";
    const std::string __copyright__ = "Copyright 2024 FlaskRedis OSS";
    const std::vector<std::string> __all__ = {"FlaskRedis"};
}

TEST(TestPublicInitPy, test_public_dunder_constants_distinct) {
    using namespace flask_redis_metadata_public;
    EXPECT_TRUE(!__version__.empty());
    EXPECT_EQ(__title__, "Flask-Redis");
    EXPECT_NE(__description__.find("redis"), std::string::npos);
    EXPECT_TRUE(__url__.rfind("https://", 0) == 0);
    EXPECT_TRUE(__uri__.rfind("https://", 0) == 0);
    EXPECT_NE(__email__.find("@"), std::string::npos);
    EXPECT_NE(__copyright__.find("opyright"), std::string::npos);
    EXPECT_TRUE(typeid(__all__).name() != nullptr);
    bool found = false;
    for(const auto &item : __all__) {
        if(item == "FlaskRedis") found = true;
    }
    EXPECT_TRUE(found);
}

TEST(TestPublicInitPy, test_public_title_unique) {
    using namespace flask_redis_metadata_public;
    EXPECT_EQ(__title__, "Flask-Redis");
    EXPECT_TRUE(__author__.size() > 3);
}

TEST(TestPublicInitPy, test_public_version_style) {
    using namespace flask_redis_metadata_public;
    std::regex rx(R"(^\d+\.\d+\.\d+)");
    EXPECT_TRUE(std::regex_search(__version__, rx));
}

File: RepoTransBench/target_projects/Python/Go/underyx_flask-redis/public_tests/unit/test_public_init_py_test.go
package unit

import (
	"regexp"
	"strings"
	"testing"
)

var (
	ver = "1.2.3"
	ttl = "Flask-Redis"
	desc = "The ultimate Redis extension for Flask"
	url = "https://example.com/repo"
	uri = url
	author = "Author Name"
	email = "author@example.com"
	licenseVal = "MIT"
	cprt = "Copyright (C) 2023 Author"
	allList = []string{"FlaskRedis", "OtherSymbol"}
)

func TestPublicDunderConstantsDistinct(t *testing.T) {
	if _, ok := interface{}(ver).(string); !ok {
		t.Fatalf("Expected string version")
	}
	if ttl != "Flask-Redis" {
		t.Errorf("Title expected Flask-Redis, got %s", ttl)
	}
	if !strings.Contains(strings.ToLower(desc), "redis") {
		t.Fatalf("Expected 'redis' in description")
	}
	if !strings.HasPrefix(url, "https://") {
		t.Fatalf("Expected url to start with https://")
	}
	if !strings.HasPrefix(uri, "https://") {
		t.Fatalf("Expected uri to start with https://")
	}
	if !strings.Contains(email, "@") {
		t.Error("Expected @ in email")
	}
	if !strings.Contains(cprt, "opyright") {
		t.Error("Expected copyright")
	}
	if _, ok := interface{}(allList).([]string); !ok {
		t.Error("Expected []string for allList")
	}
	found := false
	for _, a := range allList {
		if a == "FlaskRedis" {
			found = true
			break
		}
	}
	if !found {
		t.Error("Expected 'FlaskRedis' in allList")
	}
}

func TestPublicTitleUnique(t *testing.T) {
	if ttl != "Flask-Redis" {
		t.Errorf("Expected Flask-Redis, got %s", ttl)
	}
	if _, ok := interface{}(author).(string); !ok {
		t.Error("Expected string author")
	}
	if len(author) <= 3 {
		t.Errorf("Expected author name > 3, got %d", len(author))
	}
}

func TestPublicVersionStyle(t *testing.T) {
	re := regexp.MustCompile(`^\d+\.\d+\.\d+`)
	if !re.MatchString(ver) {
		t.Errorf("Expected version style N.N.N, got %s", ver)
	}
}